src/App.tsx

```diff
@@ -1,26 +1,254 @@
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+import { FilterStatus } from './utils/FilterStatus';
+import { Footer } from './components/Footer';
+import { Header } from './components/Header';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoList } from './components/TodoList';
+import { TodoListContext } from './variables/LangContext';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState(FilterStatus.All);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [errorVisible, setErrorVisible] = useState(false);
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [title, setTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage('Unable to load todos');
+        setErrorVisible(true);
+        setTimeout(() => {
+          setErrorVisible(false);
+        }, 3000);
+      });
+  }, []);
+
+  function handleTitleChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setTitle(event.target.value);
+  }
+
+  const reset = () => {
+    setTitle('');
+  };
+
+  function addTodo({ userId, completed }: Todo) {
+    return todoService
+      .createTodo({ userId, title, completed })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        setErrorVisible(true);
+        setTimeout(() => {
+          setErrorVisible(false);
+        }, 3000);
+        throw error;
+      });
+  }
+
+  function deleteTodo(todoId: number) {
+    setLoadingTodoIds(prevIds => [...prevIds, todoId]);
+
+    return todoService
+      .deleteTodo(todoId)
+      .then(() => {
+        setTodos(currentTodos =>
+          currentTodos.filter(todo => todo.id !== todoId),
+        );
+        if (inputRef.current) {
+          inputRef.current.focus();
+        }
+
+        setLoadingTodoIds(prevIds => prevIds.filter(id => id !== todoId));
+      })
+      .catch(error => {
+        setLoadingTodoIds(prevIds => prevIds.filter(id => id !== todoId));
```

Comment:
do not repeat the same in then and catch, just move to finally and call only once